libraries/Microsoft.Bot.Builder.Dialogs/Prompts/ChoicePrompt.cs

```diff
@@ -50,6 +38,18 @@ public ChoicePrompt(string dialogId, PromptValidator<FoundChoice> validator = nu
             DefaultLocale = defaultLocale;
         }
 
+        /// <summary>
+        /// Gets or sets a dictionary used for custom prompt display based on user-provided locale settings.
+        /// </summary>
+        /// <value>The dictionary used for custom prompt display based on user-provided locale settings.</value>
+        /// <remarks>
+        /// If set, this replaces DefaultChoiceOptions. Because this is static, it is not thread-safe.
+        /// Usage: ChoicePrompt.CustomLocaleOptions = new Dictionary<string, ChoiceFactoryOptions>()
+        ///         { "en-US", new ChoiceFactoryOptions {...} }
+        /// </remarks>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("StyleCop.CSharp.DocumentationRules", "SA1629:Documentation text should end with a period", Justification = "Doesn't like `Dictionary<string`")]
+        public static Dictionary<string, ChoiceFactoryOptions> CustomLocaleOptions { get; set; }

```

Comment:
This has implications beyond the obvious. By making this dictionary public, we cannot make any threading and concurrency assumptions about it. I would *far* prefer the dictionary itself to remain private, and only allow accessors to get/set. That way we can be correct with regards to locks and concurrency.

Reply:
I should add that your code comment helps, but I would like to see something more clean.

Reply:
Ah yeah. I should know better than that. Will fix both of those today.

Reply:
@cleemullins Changes applied, but API Compatibility Validation is throwing errors because adding an optional parameter to the existing ctor vs. adding an override. Do we want to manually bypass this or add the override?